MidiTools/src/com/mobileer/miditools/synth/SynthEngine.java

```diff
@@ -54,15 +53,23 @@
     private MidiFramer mFramer;
     private MidiReceiver mReceiver = new MyReceiver();
     private SimpleAudioOutput mAudioOutput;
+    private int mSampleRate = 48000;
```

Comment:
Glenn says the rate may change before the track is opened. The best way to get the sample rate is to use an AudioTrack.Builder and not specify a rate. Then query the AudioTrack for the rate. I will do that.